packages/core/strapi/lib/services/entity-validator/index.js

```diff
@@ -247,9 +247,14 @@ const createValidateEntity =
  * @returns {Object}
  */
 const buildRelationsStore = ({ uid, data }) => {
+  if (!uid) {
+    throw new ValidationError(`cannot build relations store: "uid" is undefined`);
```

Comment:
I'm not sure this error message will be helpful for a user. if it is realted to missing __component in the DZ then I would recommend catching this earlier to add better errors. (We can do that in another PR)

Reply:
I would suggest to have meaningful error message at every level of our code (maybe until we migrate to TS ? As TS build would fail if some mandatory parameter aren't provided).
Here we can maybe just throw `A model uid is required to build relations` ?

Reply:
I wasn't clear enough. We should keep this error but add another one earlier that is more context based with DZ. I don't think we have to wait for TS to do this.

Reply:
For sure ! More meaningful errors 🎉